reframe/core/meta.py

```diff
@@ -76,6 +76,20 @@ def __getitem__(self, key):
                             'body is disallowed'
                         ) from None
                     else:
+                        # As the last resource, look if key is a variable in
+                        # any of the base classes. If so, make its value
+                        # available in the current class' namespace.
+                        for b in self['_rfm_bases']:
+                            if (hasattr(b, '_rfm_var_space') and
+                                key in b._rfm_var_space):
+                                    v = b._rfm_var_space[key]
+                                    v.__set_name__(self, key)
+
+                                    # Store a deep-copy of the variable's
+                                    # value and return.
+                                    self._namespace[key] = v.default_value
+                                    return self._namespace[key]
```

Comment:
I think you should decrease the identation level here.

Reply:
Since this is the only place that needs to prematurely access the base classes, I've changed this a bit and filtered out the base classes that don't have a var space before they're added into this namespace. So now the `if` above is a bit simpler.